packages/core/utils/src/dml/relations/base.ts

```diff
@@ -22,14 +23,16 @@ export abstract class BaseRelationship<T> implements RelationshipType<T> {
 
   constructor(referencedEntity: T, options: Record<string, any>) {
     this.#referencedEntity = referencedEntity
-    this.#options = options
+    this.options = options
   }
 
   /**
-   * Apply nullable modifier on the schema
+   * Define the property name on the related entity that
+   * defines the inverse of this relationship.
    */
-  nullable() {
-    return new NullableModifier<T>(this)
+  mappedBy(property: string): this {
```

Comment:
nit: I might just put the mappedBy as an argument of the relation but no strong opinion

Reply:
@adrien2p, do you mean something like `belongsTo(() => User, { mappedBy: "emails" })` ?

If so, I think I agree – the strict coupling between the concepts makes sense. In case we add more chainable APIs, we will avoid something like this:
```
user: belongsTo(()) => user).somethingElse().mappedBy()
```

where the `mappedBy` is chained separately from the relationship. I know this is probably a farfetched example, but maybe something like this could happen.

Wdyt?

Reply:
yes I was thinking more like `belongsTo(() => User,  "emails")` but your example works nicely too

Reply:
I have mixed feels about this. Only because I can't tell what the difference between these two are:

```
const email = model.define("email", {
  email: model.text().optional(),
  user: model.belongsTo(() => user, { mappedBy: "emails" }).optional(),
})
```

what makes optional chainable here, but not mappedBy?

Reply:
Because optional is an option of the relation while the mapped refer to a property of the schema you are referring too. Therefore it can be almost read as user.emails

Reply:
`mappedBy` cannot live without `belongsTo()` while `optional()` is a standalone concept and can live on its own

Reply:
Both are equally correct. It all boils down to what you like more subjectively.

I'll go with the popular opinion here 😇

Reply:
That makes sense, thanks guys!

Reply:
Done!